js/blocks/loader/index.js

```diff
@@ -31,7 +31,7 @@ const registerBlocks = () => {
 		// Register the block.
 		registerBlockType( blockName, {
 			title: block.title,
-			category: block.category,
+			category: block.category.slug,
```

Comment:
What do you think about checking the type here?

```diff
diff --git a/js/blocks/loader/index.js b/js/blocks/loader/index.js
index 92d6120..e779c1e 100644
--- a/js/blocks/loader/index.js
+++ b/js/blocks/loader/index.js
@@ -31,7 +31,7 @@ const registerBlocks = () => {
                // Register the block.
                registerBlockType( blockName, {
                        title: block.title,
-                       category: block.category.slug,
+                       category: 'object' === typeof block.category ? block.category.slug : block.category,
                        icon: icon,
                        keywords: block.keywords,
                        attributes: blockAttributes( block ),
```

Without that, there were errors in my console, like:

```
The block "block-lab/range-controls" must have a registered category.
```

That error only appeared with blocks created before this PR. 

It looks like the new logic for saving categories only runs on clicking 'Update' or 'Save' in the Block Lab UI. So when I clicked 'Update' there, the console error didn't appear for the block:

<img width="1055" alt="logic-runs-here" src="https://user-images.githubusercontent.com/4063887/60405395-7e107f00-9b75-11e9-888d-b3811180c095.png">

Reply:
The need to click 'Update' has also been an issue with a few of my changes, like with how the User control changed from storing a simple integer to an object.

*Eventually*, we might think about having a [plugin update action](https://github.com/ampproject/amp-wp/pull/1487/files#diff-1a37f9f05bb7060179a77d7c2298231bR98) like Weston created. 

Then, its [callbacks](https://github.com/ampproject/amp-wp/pull/1487/files#diff-1a37f9f05bb7060179a77d7c2298231bR133) could apply changes like the one above, and users wouldn't have to click 'Update' for the new change.

Then, these changes could possibly be in its own class, like `Backwards_Compatibility`. 

Still, this would probably add complexity. It'd just move the backwards-compatibility code to one place.

Reply:
The type check is a great idea. I agree that we should have a plugin update action, and a `Backwards_Compatibility` or `Updates` class at some point in the future.

Reply:
Implemented in https://github.com/getblocklab/block-lab/pull/357/commits/b22cdfd24a0d10d5f78745317e3cee182c339077.